### ISSUE TITLE
Codespaces fails to recognize plan-name parameter when starting

### ISSUE THREAD

**Issue by justinytchen** [MicrosoftDocs/vscodespaces#702]
When passing the plan-name argument to `codespace start`, it fails to recognize the plan parameter and re-prompts the user to select it.
![image](https://user-images.githubusercontent.com/6785106/88985669-bfe28c00-d285-11ea-8b8c-a26a3a902b12.png)

**Comment by jkeech** [comment]
I think it requires the fully-qualified azure resource id such as `/subscriptions/{id}/...../{plan}`. Can you retry that? @taiyosogawa, I think we could improve both the docs and the CLI output in this case.

**Comment by justinytchen** [comment]
It does recognize the fully qualified, but then it prompts for the resource group name, which is in the qualified ID.

**Comment by jkeech** [comment]
Yeah, that doesn't really make sense. Seems like something that would be easy to improve. Thanks for the feedback!

**Comment by taiyosogawa** [comment]
@justinytchen, thanks for reporting this. It prompts you to select the plan (even though there is only 1) because the CLI now has the ability to create a new plan for you. If you pressed Enter without entering '1', it would think that you wanted to create a new plan.

To echo @jkeech, we'll take this feedback into consideration for improving the clarity of the CLI experience.

**Comment by taiyosogawa** [comment]
I've made some changes to the help text to indicate that the `--plan-name` and `--resource-group` can be either existing or new. (Will roll out next week). We'll also use this issue to help inform how to refine the actual mechanics of the CLI. I'll close the issue for now, but please let us know if your question was not answered.

**Comment by justinytchen** [comment]
> @justinytchen, thanks for reporting this. It prompts you to select the plan (even though there is only 1) because the CLI now has the ability to create a new plan for you. If you pressed Enter without entering '1', it would think that you wanted to create a new plan.

Sorry, I'm a little confused. Would it need to still prompt me to select a plan if I passed in the plan-name as an argument? I can see why it would prompt me even if there was only 1 plan, under the assumption that no command line args were present, but in the above example, the plan name is passed in after the -p flag. 

Also in response to
> I think it requires the fully-qualified azure resource id such as `/subscriptions/{id}/...../{plan}`. Can you retry that? 
@TylerLeonhardt has used the same plan-name flag without the fully qualified ID(i.e. just vscs-plan-westus2). Are there different cases where the full ID is required?

**Comment by jkeech** [comment]
> @TylerLeonhardt mentioned that he used the same plan-name flag without the fully qualified ID(i.e. just vscs-plan-westus2). Are there different cases where the full ID is required?

I think the intent is that you either specify the subscription/resource group/plan name individually, OR you specify the fully-qualified resource Id which contains all of that info. You should not need to do both, but it sounds like there is a bug.

**Comment by TylerLeonhardt** [comment]
probably should reopen this then, I think.

**Comment by Davsterl** [comment]
We'll take a look!

**Comment by bchuecos** [comment]
I think there's a bit of confusion on how the flags work and which flags are required together.

The --plan-name / -p flag specifies the plan name (in OP's case, vscs-plan-westus2).
The -p flag by itself doesn't uniquely identify your plan name, you have to provide a --resource-group / -r. This is because you could potentially have multiple plans with the same name, but under different resource groups.

So in OP's case, to avoid having the CLI prompt you for the plan, you'd have to do:
`./bin/codespaces start -s <subscription> -r <resource group> -p vscs-plan-westus2`

Alternatively, you can use the --plan-id / -i flag, in which you would specify the full plan ID (subscriptions/...). This way, you don't have to specify the resource group since it's already included in the path:
`./bin/codespaces start -s <subscription> -i  /subscriptions/{id}/resourceGroups/{resource group}/...../{plan}`

Hope this helps!

**Comment by TylerLeonhardt** [comment]
Yeah I guess the error message could be improved to say you need to specify a resource group if you specify a plan name.
